### PR TITLE
Admin have no access to templates

### PR DESCRIPTION
**Pull request by DimaSmagulov**
code review:
- [x] @CHILL99 
- [ ] @krisberry 
- [x] @natalya5 
- [x] @oleg08 
- [x] @OrestF 
- [ ] @rtershak 
- [x] @vvs79 

2nd lvl review:

- [ ] @Kmalynss
- [x] @serhiym 
- [ ] @ruba-ruba

https://trello.com/c/OpfZOsP1/27-ensure-that-guest-user-not-logged-in-has-no-access-to-templates-functionality This task have been comleted earlie, so i just rewoked it to enshure that admin has no access to templates.

1. Admin has no access to templates
2. Styled templates pages
3. Some fixes

### CODE REVIEW

File: app/controllers/admin/topics_controller.rb [hunk 1]
```diff
@@ -61,9 +61,6 @@ def destroy
       @topic.destroy
       flash[:success] = 'Topic deleted'
       redirect_to admin_topics_path
-    else 
-      flash[:danger] = 'You can not delete topic with questions!'
-      redirect_to admin_topics_path  
     end
```
Comment (OrestF) [inline app/controllers/admin/topics_controller.rb]: Why you remove this condition?

Comment (OrestF) [inline app/controllers/admin/topics_controller.rb]: Why do you remove this condition?

File: app/assets/stylesheets/custom.css.scss [hunk 2]
```diff
@@ -133,13 +129,13 @@ input, textarea, select, .uneditable-input {
...
     color: inherit;
-    i{
+    i {
       color: white;
     }
   }
 }
 
-form{
+form {
```
Comment (vvs79) [inline app/assets/stylesheets/custom.css.scss]: 123

File: app/views/admin/topics/index.html.slim [hunk 3]
```diff
@@ -8,8 +8,11 @@
...
       = link_to 'Create new topic', new_admin_topic_path, class: 'btn btn-md btn-success'
   .center
-    = will_paginate 
-  br  
+    = will_paginate
+  br
+  .col-sm-12
+    .center
+      = will_paginate 
   ul.list-unstyled
```
Comment (vvs79) [inline app/views/admin/topics/index.html.slim]: will_paginate 2 times ?

File: app/controllers/admin/topics_controller.rb (hunk 1 above)
Comment (DimaSmagulov) [inline app/controllers/admin/topics_controller.rb]: Coz we have no access to button delete when topic has questions. 
-if @questions.empty? 
        = link_to 'Delete topic', admin_topic_path(@topic), method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-danger'
so i think we don't need this condition.

File: app/views/admin/topics/index.html.slim (hunk 3 above)
Comment (DimaSmagulov) [inline app/views/admin/topics/index.html.slim]: oops) thx dude!

**Comment by vvs79** [comment]
reviewed

File: app/views/admin/topics/index.html.slim (hunk 3 above)
Comment (DimaSmagulov) [inline app/views/admin/topics/index.html.slim]: fixed. thx one more time)

**Comment by natalya5** [comment]
reviewed

**Comment by oleg08** [comment]
reviewed

File: app/controllers/admin/topics_controller.rb (hunk 1 above)
Comment (OrestF) [inline app/controllers/admin/topics_controller.rb]: Its true, but if someone use something like RESTclient. I think it will be reliable not to remove this code. But i understand your logic)

Comment (DimaSmagulov) [inline app/controllers/admin/topics_controller.rb]: Need to read about it.

File: app/controllers/templates_controller.rb [hunk 4]
```diff
@@ -59,4 +66,10 @@ def destroy
   def template_params
     params.require(:template).permit(:name, :question_ids => [])
   end
+
+  def check_admin
```
Comment (Kmalynss) [inline app/controllers/templates_controller.rb]: Why did you add this back? Templates is something that interviewer should manage? This already was removed in one of previous commits.

File: app/views/templates/_form.html.slim [hunk 5]
```diff
@@ -1,8 +1,16 @@
 .form-group
-  = form_for(@template) do |f|
+  = form_for [@template], html: { multipart: true } do |f|
```
Comment (Kmalynss) [inline app/views/templates/_form.html.slim]: Why did you warp ```@template``` with ```[...]```?

File: app/controllers/admin/topics_controller.rb (hunk 1 above)
Comment (serhiym) [inline app/controllers/admin/topics_controller.rb]: @DimaSmagulov - just add code back - in fact we won't use this condition but code should let's be consistent
(there is 0.001% possibility that we can will call this action from other view in future)
@OrestF - for "rest client"  this code should be changed

File: app/controllers/templates_controller.rb (hunk 4 above)
Comment (DimaSmagulov) [inline app/controllers/templates_controller.rb]: I add this because our admin is user too, so he still have access to templates. As i understand only interviewer has acces to templates. So i check if user admin and close access to templates for him 
def check_admin
    if current_user.admin? 
      redirect_to authenticated_root_path, notice: 'Access Denied' 
    end

File: app/views/templates/_form.html.slim (hunk 5 above)
Comment (DimaSmagulov) [inline app/views/templates/_form.html.slim]: fixed it and we dont need html: { multipart: true } for this form

File: app/controllers/admin/topics_controller.rb (hunk 1 above)
Comment (DimaSmagulov) [inline app/controllers/admin/topics_controller.rb]: ok fixed it

**Comment by CHILL99** [comment]
reviewed

File: app/controllers/templates_controller.rb (hunk 4 above)
Comment (Kmalynss) [inline app/controllers/templates_controller.rb]: :+1:
